### PR TITLE
fix(SSRF): Add missing validation to routes in portal and gef

### PR DESCRIPTION
**Pull request by JennyHumphries**
### Introduction
This review is to clear the last 15 SSRF warnings [here](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning?query=is%3Aopen+branch%3Amain+rule%3Ajssecurity%3AS5144+).

### Resolution
* Added validation for mongo ids in `portal` (Warnings [427](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/427), [431](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/431), [439](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/439))
* Added validation for mongo ids in `gef-ui` (Warnings [391](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/391), [392](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/392), [393](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/393), [418](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/418), [419](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/419), [420](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/420), [421](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/421))
* Added postcode validation in `gef-ui` (Warning [394](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/394))
* Added companies house number validation in `gef-ui` (Warning [422](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/422))
* Added reset password token validation in`gef-ui` (Warning [441](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/441))
    * We check that it's a hexadecimal 
* Add validation for fieldname and filename (Warning [435](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/435))
    * for fieldname we check against a list of allowed values
    * for filename we check that it's of the format [sometext].[allowed extension]
* Remove "origin" from endpoint (Warning [446](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning/446))
* I've also checked `api.js` for any other endpoints that look potentially unprotected, and added a couple of extra `isMongoId` checks

### To Do
Make sure all tests pass

### CODE REVIEW

File: gef-ui/server/services/__tests__/api.test.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ afterEach(() => {
   jest.clearAllMocks();
 });
 
+const fakeMongoId = 'ABCDEF123456789000000000';
```
Comment (ChristianMcc) [inline gef-ui/server/services/__tests__/api.test.js:11]: Could we add in some tests to check false is returned from all these newly validated endpoints with the following mongo IDs: '../../../etc/passwd/', 'localhost', '{}', '[]', ''.

Might be worth just making a list of these as inputs and doing a test.each loop for each of the endpoints, something like:

```
const invalidFacilityUrnTestCases = [["123"], ["127.0.0.1"], ["{}"], ["[]"]];

describe("when application ID is invalid", () => {
  test.each(invalidMongoId)(
    "returns a 400 if you provide an invalid application ID: %s",
    async (facilityUrn) => {

      const response = await api.getApplication(invalidMongoId);

      expect(response).toEqual(false);
    }
  );
});
```

File: portal/server/constants/regex.js [hunk 2]
```diff
@@ -1 +1,5 @@
+const { FILE_UPLOAD } = require('./file-upload');
+
 exports.COMPANIES_HOUSE_NUMBER_REGEX = /^(([A-Z]{2}|[A-Z]\d{1}|\d{2})(\d{5,6}|\d{4,5}[A-Z]))$/;
+
+exports.FILE_NAME_REGEX = new RegExp(`\\.(${FILE_UPLOAD.ALLOWED_FORMATS.join('|')})$`);
```
Comment (ChristianMcc) [inline portal/server/constants/regex.js:5]: Can this be changed to need at least 1 character before the .<FILE_EXTENSION> please

**Review by ChristianMcc — COMMENTED** [review]
Nice work, minor comments which I can take care of tomorrow :)

File: portal/server/api.js [hunk 3]
```diff
@@ -18,13 +17,15 @@ const login = async (username, password) => {
...
     });
 
-    return response.data ? {
-      success: response.data.success,
-      token: response.data.token,
-      user: response.data.user,
-    } : '';
+    return response.data
+      ? {
+          success: response.data.success,
```
Comment (abhi-markan) [inline portal/server/api.js:22]: Various linting errors to resolve.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Lint error to resolve please.

**Review by IgnatG — APPROVED** [review]

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**